spec/s3/app_spec.rb

```diff
@@ -0,0 +1,64 @@
+require_relative "../spec_helper"
+
+describe "S3 provider" do
+  def container_url_for(user)
+    "#{app.settings.s3["endpoint"]}#{app.settings.s3["bucket"]}/#{user}"
+  end
+
+  def storage_class
+    RemoteStorage::S3
+  end
+
+  before do
+    stub_request(:put, "#{container_url_for("phil")}/food/aguacate").
+      to_return(status: 200, headers: { etag: '"0815etag"' })
+    # Write new content with an If-Match header (a new Etag is returned)
+    stub_request(:put, "#{container_url_for("phil")}/food/aguacate").
+      with(body: "aye").
+      to_return(status: 200, headers: { etag: '"0915etag"' })
+    stub_request(:head, "#{container_url_for("phil")}/food/aguacate").
+      to_return(status: 200, headers: { last_modified: "Fri, 04 Mar 2016 12:20:18 GMT" })
+    stub_request(:get, "#{container_url_for("phil")}/food/aguacate").
+      to_return(status: 200, body: "rootbody", headers: { etag: '"0817etag"', content_type: "text/plain; charset=utf-8" })
+    stub_request(:delete, "#{container_url_for("phil")}/food/aguacate").
+      to_return(status: 200, headers: { etag: '"0815etag"' })
+
+    # PUT requests authorized updates the metadata object in redis when it changes
```

Comment:
This looks like it's copied from test output, but it sure isn't English language that one can understand.